content/en/logs/guide/google-cloud-log-forwarding.md

```diff
@@ -0,0 +1,294 @@
+
+{{% collapse-content title="Manual" level="h4" id="manual-logging-setup" %}}
+The instructions in this section guide you through the process of:
+
+1. Creating a Pub/Sub [topic][401] and [pull subscription][402] to receive logs from a configured log sink
+2. Creating a custom Dataflow worker service account to provide [least privilege][403] to your Dataflow pipeline workers
+3. Creating a [log sink][404] to publish logs to the Pub/Sub topic
+4. Creating a Dataflow job using the [Datadog template][400] to stream logs from the Pub/Sub subscription to Datadog
+
+You have full control over which logs are sent to Datadog through the logging filters you create in the log sink, including GCE and GKE logs. See Google's [Logging query language page][405] for information about writing filters. For a detailed examination of the created architecture, see [Stream logs from Google Cloud to Datadog][406] in the Cloud Architecture Center.
+
+**Note**: You must enable the **Dataflow API** to use Google Cloud Dataflow. See [Enabling APIs][407] in the Google Cloud documentation for more information.
+
+To collect logs from applications running in GCE or GKE, you can also use the [Datadog Agent][408].
+
+#### 1. Create a Cloud Pub/Sub topic and subscription
+
+1. Go to the [Cloud Pub/Sub console][409] and create a new topic. Select the option **Add a default subscription** to simplify the setup.
+
+   **Note**: You can also manually configure a [Cloud Pub/Sub subscription][410] with the **Pull** delivery type. If you manually create your Pub/Sub subscription, leave the `Enable dead lettering` box **unchecked**. For more details, see [Unsupported Pub/Sub features][411].
+
+{{< img src="integrations/google_cloud_platform/create_a_topic.png" alt="The Create a topic page in the Google Cloud Console with the Add a default subscription checkbox selected" style="width:80%;">}}
+
+2. Give that topic an explicit name such as `export-logs-to-datadog` and click **Create**.
+
+3. Create an additional topic and default subscription to handle any log messages rejected by the Datadog API. The name of this topic is used within the Datadog Dataflow template as part of the path configuration for the `outputDeadletterTopic` [template parameter][412]. When you have inspected and corrected any issues in the failed messages, send them back to the original `export-logs-to-datadog` topic by running a [Pub/Sub to Pub/Sub template][413] job.
+
+4. Datadog recommends creating a secret in [Secret Manager][414] with your valid Datadog API key value, for later use in the Datadog Dataflow template.
+
+**Warning**: Cloud Pub/Subs are subject to [Google Cloud quotas and limitations][415]. If the number of logs you have exceeds those limitations, Datadog recommends you split your logs over several topics. See [the Monitor the Pub/Sub Log Forwarding section](#monitor-the-cloud-pubsub-log-forwarding) for information on setting up monitor notifications if you approach those limits.
```

Comment:
This is the yellow alert box. Should this be at the top of this section instead of the bottom? 

```suggestion
<div class="alert alert-danger">
Cloud Pub/Subs are subject to <a href="https://cloud.google.com/pubsub/quotas#quotas">Google Cloud quotas and limitations</a>. If the number of logs you have exceeds those limitations, Datadog recommends you split your logs over several topics. See the <a href="#monitor-the-cloud-pubsub-log-forwarding">Monitor the Pub/Sub Log Forwarding</a> section for information on setting up monitor notifications if you approach those limits.
</div>
```